nullaway/src/main/java/com/uber/nullaway/fixserialization/Serializer.java

```diff
@@ -195,4 +203,116 @@ public static Path pathToSourceFileFromURI(@Nullable URI uri) {
       return path;
     }
   }
+
+  /**
+   * Serializes the given {@link Symbol} to a string.
+   *
+   * @param symbol The symbol to serialize.
+   * @return The serialized symbol.
+   */
+  public static String serializeSymbol(@Nullable Symbol symbol) {
+    if (symbol == null) {
+      return "null";
+    }
+    switch (symbol.getKind()) {
+      case FIELD:
+      case PARAMETER:
+        return symbol.name.toString();
+      case METHOD:
+      case CONSTRUCTOR:
+        return serializeMethodSignature((Symbol.MethodSymbol) symbol);
+      default:
+        return symbol.flatName().toString();
+    }
+  }
+
+  /**
+   * Serializes the signature of the given {@link Symbol.MethodSymbol} to a string.
+   *
+   * @param methodSymbol The method symbol to serialize.
+   * @return The serialized method symbol.
+   */
+  private static String serializeMethodSignature(@Nullable Symbol.MethodSymbol methodSymbol) {
+    if (methodSymbol == null) {
+      return "null";
+    }
+    StringBuilder sb = new StringBuilder();
+    if (methodSymbol.isConstructor()) {
+      // For constructors, we use the name of the enclosing class.
+      Name name = methodSymbol.owner.enclClass().getSimpleName();
+      if (name.isEmpty()) {
+        name = methodSymbol.owner.enclClass().getSuperclass().asElement().getSimpleName();
+      }
+      sb.append(name);
+    } else {
+      // For methods, we use the name of the method.
+      sb.append(methodSymbol.getSimpleName());
+    }
+    // Index for iteration in visiting method parameters. Used to determine if visiting the last
+    // parameter.
+    AtomicInteger index = new AtomicInteger(1);
```

Comment:
I prefer just using a loop with an index to a stream with an `AtomicInteger` keeping the index.  We could just directly update `sb` as we go.  To me, that would make for clearer code

Reply:
This is also removed as part of d74cde16a2d44e830d1b5caa52d1efdb9ba3edd6.